src/VisualStudio/CSharp/Impl/LanguageService/CSharpCodeCleanupFixer.cs

```diff
@@ -0,0 +1,348 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.ComponentModel.Composition;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeCleanup;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.RemoveUnusedVariable;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editor;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Shared.Utilities;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.VisualStudio.Editor.CodeCleanup;
+using Microsoft.VisualStudio.Language.CodeCleanUp;
+using Microsoft.VisualStudio.LanguageServices.Implementation.CodeCleanup;
+using Microsoft.VisualStudio.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServices.CSharp.LanguageService
+{
+    [Export(typeof(CodeCleanUpFixer))]
+    [ContentType(ContentTypeNames.CSharpContentType)]
+    internal class CSharpCodeCleanUpFixer : CodeCleanUpFixer
+    {
+        private const string RemoveUnusedImportsFixId = "RemoveUnusedImportsFixId";
+        private const string SortImportsFixId = "SortImportsFixId";
+
+        [Export]
+        [FixId(IDEDiagnosticIds.UseImplicitTypeDiagnosticId)]
+        [Name(IDEDiagnosticIds.UseImplicitTypeDiagnosticId)]
+        [Order(After = IDEDiagnosticIds.UseExpressionBodyForConstructorsDiagnosticId)]
+        [ConfigurationKey("unused")]
+        [HelpLink("https://www.microsoft.com")]
+        [LocalizedName(typeof(CSharpFeaturesResources), nameof(CSharpFeaturesResources.Apply_implicit_explicit_type_preferences))]
```

Comment:
Let's improve this post-Preview1. Mapping diagnostic id to a fix id one to one seems wrong as you can see in these repeating exports.

Reply:
this being static is a problem for us. since our fixers can be added on runtime (dynamic).

having a static way to declare it fine, but we need some way to be dynamic as well.

Reply:
that needs some design as users would want to configure which fixes they want to run so they need to be known ahead of time.